sike/sikep751/sikep751.go

```diff
@@ -0,0 +1,168 @@
+// This file contains APIs for Supersingular Isogeny Key Encapsulation (SIKE) protocol
+// which is recently submitted to NIST PQC standardization workshop.
+// The underlying core functions are based on SIDH API developed by Cloudflare.
+// SIKE requires NIST's approved hash functions such as sha3 and cSHAKE256 to encapsulate
+// the key based on NIST specifications. The sha3 package along with cSHAKE256 implementation
+// is placed inside the p751sidh package. Note that the standard sha3 package from
+// "golang.org/x/crypto/sha3" does not include cSHAKE256 functions.
+//
+// Author: Amir Jalali  				ajalali2016@fau.edu
+// Date: Feb 2018
+package sikep751
+
+import (
+	"bytes"
+	"io"
+)
+
+import . "github.com/cloudflare/p751sidh/sha3"
+
+import . "github.com/cloudflare/p751sidh/sidh/sidhp751"
+
+// SIKE implementation parameters correspond to p751
+const (
+	// MessageSize in bytes.
+	MessageSize = 32
+	// CiphertextSize in bytes = SIDHPublicKeySize + MessageSize
+	CiphertextSize = 596
+	// SIKESecretKeySize in bytes = MessageSize + SIDHPublicKeySize + SIDHSecretKeySize
+	SIKESecretKeySize = 644
+	// G is a custom vallue for cSHAKE256
+	G = "0"
```

Comment:
As per SIKE submission to PQ NIST (see 1.4) those values should be
G = 0x00 (not 0x30)
H = 0x01 (not 0x31)
F = 0x02 (not 0x32)